ext/AbstractDifferentiationEnzymeExt.jl

```diff
@@ -0,0 +1,62 @@
+module AbstractDifferentiationEnzymeExt
+
+if isdefined(Base, :get_extension)
+    import AbstractDifferentiation as AD
+    using Enzyme: Enzyme
+else
+    import ..AbstractDifferentiation as AD
+    using ..Enzyme: Enzyme
+end
+
+struct Mutating{F}
+    f::F
+end
+function (f::Mutating)(y, xs...)
+    y .= f.f(xs...)
+    return y
+end
+
+AD.@primitive function value_and_pullback_function(b::AD.EnzymeReverseBackend, f, xs...)
+    y = f(xs...)
+    return y,
+    Δ -> begin
+        Δ_xs = zero.(xs)
+        dup = if y isa Real
+            if Δ isa Real
+                Enzyme.Duplicated([y], [Δ])
+            elseif Δ isa Tuple{Real}
+                Enzyme.Duplicated([y], [Δ[1]])
+            else
+                throw(ArgumentError("Unsupported cotangent type."))
+            end
+        else
+            if Δ isa AbstractArray{<:Real}
+                Enzyme.Duplicated(y, Δ)
+            elseif Δ isa Tuple{AbstractArray{<:Real}}
+                Enzyme.Duplicated(y, Δ[1])
+            else
+                throw(ArgumentError("Unsupported cotangent type."))
+            end
+        end
+        Enzyme.autodiff(
+            Enzyme.Reverse,
+            Mutating(f),
+            Enzyme.Const,
+            dup,
+            Enzyme.Duplicated.(xs, Δ_xs)...,
```

Comment:
That means users of AbstractDifferentiation miss a major feature of Enzyme. But maybe it's unavoidable and the current design of AbstractDifferentiation can't support it and the wrapper will always be less performant than Enzyme?

Reply:
let's brainstorm solutions, I think it's possible to support partial pullback with an extended API